video/out/opengl/hwdec_cuda.c

```diff
@@ -81,19 +101,44 @@ static int cuda_init(struct ra_hwdec *hw)
     CUdevice display_dev;
     AVBufferRef *hw_device_ctx = NULL;
     CUcontext dummy;
-    unsigned int device_count;
     int ret = 0;
     struct priv_owner *p = hw->priv;
     CudaFunctions *cu;
 
-    if (!ra_is_gl(hw->ra))
-        return -1;
+#if HAVE_GL
+    bool is_gl = ra_is_gl(hw->ra);
+#endif
 
-    GL *gl = ra_gl_get(hw->ra);
-    if (gl->version < 210 && gl->es < 300) {
-        MP_VERBOSE(hw, "need OpenGL >= 2.1 or OpenGL-ES >= 3.0\n");
-        return -1;
+#if HAVE_VULKAN
+    bool is_vk = ra_vk_get(hw->ra) != NULL;
+#endif
+
+#if HAVE_GL
+    if (is_gl) {
+        GL *gl = ra_gl_get(hw->ra);
+        if (gl->version < 210 && gl->es < 300) {
+            MP_VERBOSE(hw, "need OpenGL >= 2.1 or OpenGL-ES >= 3.0\n");
+            return -1;
+        }
     }
+#endif
+#if HAVE_VULKAN
+    if (is_vk) {
+    #if HAVE_WIN32_DESKTOP
```

Comment:
You define the API type of the RA_BUF_SHARED_MEMORY depending on the platform anyway, so only one will be useful - do we really need two separate fields for has_ext_external_memory_win32/fd if the user can only use one or the other depending on the platform? That would eliminate this sort of awkward branch.

Reply:
has_ext_external_memory_fd does exist and work on Windows, it's just not used for CUDA. Those variables match the vulkan extension names.

Reply:
Incidentally, you can also guard against the availability of the extension in vulkan.h, using `#ifdef VK_KHR_external_memory_fd`, that would allow you to skip hard-coding assumptions about the platform. (In particular, if `VK_KHR_external_memory_win32` is defined, `HANDLE` etc. is definitely in scope)

Another possibility here would be to allow configuring the type of external memory when allocating the VkBuffer, and use a `union` for the various platform-specific handle types. That would allow using an `fd` even on windows, if something else ends up needing it - rather than hard-coding the assumption that win32 will always only use HANDLE. Then the separation of the extension names would be justified.

Reply:
Would you like me to specifically make either of these changes? I think the current state is pretty clean and we can adapt it if there's a need down the line.

Reply:
Sounds like a plan.